### PR TITLE
SOL-149404: Apply select= field filtering to all monitoring tools to reduce response size returned to the LLM

### PR DESCRIPTION
**Pull request by helen-zhu-solace**
## Summary

Reduces the size of SEMP monitoring responses sent to the LLM by requesting
only the fields each tool actually uses, via the SEMP v2 `select=` query
parameter. Also fixes the HTTP client so array-typed query parameters
(`select`, `where`) are encoded in SEMP's `collectionFormat=csv` shape
instead of being rejected by the broker.

Fixes [SOL-149404](https://sol-jira.atlassian.net/browse/SOL-149404)

## Type of Change

- [x] Performance improvement
- [x] Bug fix (non-breaking change which fixes an issue)

## Motivation and Context

Monitoring tools were returning full objects from the broker, which pushed
large, mostly-unused payloads into the LLM context and inflated token
usage. SEMP v2 supports `select=` for field filtering, but our HTTP client
was url-encoding the commas in array-typed query params (`%2C`), and the
broker rejects those with `'a,b,c' not a valid attribute`. So enabling
`select=` required fixing the encoding path first.

## Changes Made

- `internal/semp/sempv2/client.go`: split query-param construction so
  array-typed params (e.g. `select`, `where`) are pre-encoded as
  `name=v1,v2,v3` with literal commas and appended alongside the
  standard-encoded params.
- `internal/semp/sempv2/client.go`: new `encodeCSVQueryParam` helper that
  accepts `[]string`, `[]any`, or a comma-joined string, trims empties,
  and percent-encodes each element individually.
- `internal/semp/sempv2/client_test.go`: tests covering the CSV encoding
  (string / slice inputs, empty-element trimming, coexistence with
  non-array query params).
- `internal/composite/definitions/tools.yaml`: added curated `select=`
  lists to 10 monitoring tool operations (VPN, queues, clients, RDP,
  DMR cluster + links, rates, etc.), keeping only fields the tool
  actually surfaces.

## Testing

**Test Environment:**
- Go version: 1.25.0 linux/amd64
- OS: Linux (Fedora) 40
- Broker version: Solace PubSub+ Software (VMR), SEMP 100.0main.0.7549 (internal lab build)

**Tests Performed:**
- [x] Unit tests added/updated (`client_test.go`)
- [x] Tested locally with `go test -race ./...`
- [x] Tested with real Solace broker 

**Test Coverage:**
- CSV-encoded `select` values reach the broker with raw commas
- Array param coexists with non-array params in the same request
- Empty / whitespace entries in the select list are dropped
- Existing (non-array) query-param behavior is unchanged

## Breaking Changes

None. Tools return a strict subset of the fields they returned before; no
public Go API changed.

### Code Quality
- [x] Self-review completed
- [x] Comment on `buildRequest` explains *why* CSV is encoded manually
- [x] No commented-out code or debug statements

### Security
- [x] No credentials in code
- [x] No new credential-carrying types
- [x] `/check-logs` run — no new logging violations

### Testing
- [x] `go test -race ./...` passes locally
- [x] Edge cases covered (empty entries, slice vs string input)

### Documentation
- [x] README.md — N/A (no user-facing surface change)
- [x] docs/ — N/A
- [x] godoc on `encodeCSVQueryParam` explains the SEMP CSV contract

### Git & CI
- [x] Commits signed off (DCO)  
- [x] Clear commit message (`SOL-149404: ...`)
- [x] Branch up to date with main
- [x] No merge conflicts

## Additional Notes

The `select=` lists are hand-curated per tool. If a tool is extended to
show more fields in the future, the corresponding `select=` list in
`tools.yaml` must be expanded too, or the new field will silently be
absent from the response.

## Related Issues/PRs

- SOL-149343 (prior PR correcting list tool descriptions — same series)

---

**For Reviewers:**

**Focus Areas:**
- `encodeCSVQueryParam` — correctness of the manual encoding (matches
  what SEMP expects, handles all input shapes we produce from YAML).
- The `select=` field lists in `tools.yaml` — are any important fields
  missing for the use cases each tool supports?

[SOL-149404]: https://sol-jira.atlassian.net/browse/SOL-149404?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/composite/definitions/tools.yaml [hunk 1]
```diff
@@ -48,12 +48,14 @@ tools:
         args:
           msgVpnName: "{{.Params.msgVpnName}}"
           restDeliveryPointName: "{{.Params.restDeliveryPointName}}"
+          select: "lastFailureReason, lastFailureTime, postRequestTarget, queueBindingName, restDeliveryPointName, up, uptime"
       - id: restConsumers
```
[the remark this answers is not in the document]
Comment (bczoma) [reply internal/composite/definitions/tools.yaml:53]: Agree

**Review by bczoma — CHANGES_REQUESTED** [review]

File: internal/composite/definitions/tools.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (helen-zhu-solace) [reply internal/composite/definitions/tools.yaml:53]: rdpStatus step in get-rdp-status now uses a curated select= list(clientName, clientProfileName, enabled, lastFailureReason, lastFailureTime, msgVpnName, restDeliveryPointName, timeConnectionsBlocked, up).

File: internal/semp/sempv2/client_test.go [hunk 2]
```diff
@@ -134,6 +134,41 @@ func TestClient_Execute_QueryParams(t *testing.T) {
 	}
 }
 
+// TestClient_Execute_QueryParams_ArrayCSVRawCommas pins the wire format for
+// SEMP v2 array query params (select, where): a single param whose value uses
+// raw commas, not %2C. The broker treats %2C-encoded commas as part of the
+// attribute name and rejects the request with "not a valid attribute".
+func TestClient_Execute_QueryParams_ArrayCSVRawCommas(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (helen-zhu-solace) [reply internal/semp/sempv2/client_test.go:141]: Added unit tests for []string / []any inputs, whitespace-trim/empty-drop, and array+regular-param coexistence

**Review by bczoma — APPROVED** [review]
Looks good!
